app/wait_time/location.py

```diff
@@ -0,0 +1,65 @@
+from datetime import datetime, timezone
+from typing import Dict, Any
+from app.common import BadDataError
+import json
+
+
+class UserLocation:
+    def __init__(
+        self,
+        aid: str,
+        latitude: float,
+        longitude: float,
+        timestamp: datetime = datetime.now(timezone.utc),
+    ):
+        self.aid = aid
+        self.latitude = latitude
+        self.longitude = longitude
+        self.timestamp = timestamp
+
+    @staticmethod
+    def from_dict(aid: str, dict: Dict[str, Any]) -> "UserLocation":
+        """
+        Creates a new UserLocation object from a Python Dictionary
+
+        :param dict: Dictionary of key-value pairs corresponding to location.
+        :returns: UserLocation from specified data
+        :raises BadDataError: If required data is missing from the dictionary
+        """
+        try:
+            location = UserLocation(
+                aid,
+                dict["latitude"],
+                dict["longitude"],
+                dict.get("timestamp", datetime.now(timezone.utc)),
+            )
+        except KeyError as e:
+            raise BadDataError("Missing data from Location data: " + str(e))
+
+        if not _is_valid_gps_coordinates(location.latitude, location.longitude):
```

Comment:
nit: Generally it's best practice for error checks to be performed first before any computation (early return).

Reply:
Yes sir, put this here since the previous bit is needed to make sure that latitude and longitude are part of the dict first

Reply:
Ah dw bout it then.